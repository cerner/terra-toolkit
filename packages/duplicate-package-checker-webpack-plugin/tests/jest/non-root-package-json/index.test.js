const webpack = require('webpack');
const chalk = require('chalk');
const config = require('./webpack.config');

const chalkLevel = chalk.level;

describe('Dependency tree with non-root package.json', () => {
  beforeEach(() => {
    /**
     * Travis-ci doesn't support colorized string generation with Chalk. To prevent snapshot mismatches
     * with Travis's test output, Chalk is disabled for snapshot tests.
     */
    chalk.level = 0;
  });

  afterEach(() => {
    chalk.level = chalkLevel;
  });

  it('should output warnings', (done) => {
    webpack(config, (err, stats) => {
      expect(stats.compilation.warnings[0].message).toMatchSnapshot();
      done();
    });
  });
});
