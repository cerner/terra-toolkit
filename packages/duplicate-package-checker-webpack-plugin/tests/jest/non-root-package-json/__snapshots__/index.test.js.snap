// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dependency tree with non-root package.json should output warnings 1`] = `
"a
  Multiple versions of a found:
    1.0.0 ./~/a
    2.0.0 ./~/b/~/a
"
`;
